### PR TITLE
Added Share Count and Added relation to post model

### PR DESCRIPTION
**Pull request by PeterDMadrid**
**OVERVIEW**

- Share Count

### CODE REVIEW

File: src/app/Models/Post.php [hunk 1]
```diff
@@ -52,4 +52,9 @@ public function sharedPost(): BelongsTo
     {
         return $this->belongsTo(Post::class, 'childPost_id');
     }
+
+    public function numberOfShares(): hasMany
```
Comment (mikco-hipe) [inline src/app/Models/Post.php:70]: doc blocs

File: src/app/Models/Post.php [hunk 2]
```diff
@@ -52,4 +52,9 @@ public function sharedPost(): BelongsTo
...
+    {
+        return $this->hasMany(Post::class, 'childPost_id');
```
Comment (mikco-hipe) [inline src/app/Models/Post.php]: use snake case for `'childPost_id'`

**Review by mikco-hipe — COMMENTED** [review]
Please check!

**Review by hipe01mark — APPROVED** [review]
LGTM 🙇

**Pull request merged**